test/unit/httpapi_plugins/test_ftd.py

```diff
@@ -314,3 +315,96 @@ def _connection_response(response, status=200):
         response_text = json.dumps(response) if type(response) is dict else response
         response_data = BytesIO(response_text.encode() if response_text else ''.encode())
         return response_mock, response_data
+
+    def test_get_list_of_supported_api_versions_with_failed_http_request(self):
+        send_mock = mock.MagicMock(side_effect=HTTPError('url', 'code', 'msg', 'hdrs', 'fp'))
+        with mock.patch.object(self.ftd_plugin.connection, 'send', send_mock):
+            self.assertRaises(
+                ConnectionError,
+                self.ftd_plugin._get_list_of_supported_api_versions
+            )
+
+    def test_get_list_of_supported_api_versions_with_buggy_response(self):
+        http_response_mock = mock.MagicMock()
+        http_response_mock.getvalue.return_value = "Non JSON value"
+
+        send_mock = mock.MagicMock(return_value=(None, http_response_mock))
+        with mock.patch.object(self.ftd_plugin.connection, 'send', send_mock):
+            self.assertRaises(
+                ConnectionError,
```

Comment:
Let's also check that we get reasonable and actionable error messages, so users can understand what went wrong.

Reply:
fixed